client/src/components/Form/Elements/FormOptionalText.vue

```diff
@@ -75,32 +75,22 @@ export default {
                 return this.value;
             },
             set(val) {
-                this.setValue(val);
+                this.$emit("input", val);
             },
         },
         currentStatus: {
             get() {
                 return this.status;
             },
             set(val) {
-                this.status = val;
-                if (val == false) {
-                    this.currentValue = null;
-                } else {
+                this.status = Boolean(val);
+                if (this.status) {
                     this.currentValue = "";
+                } else {
+                    this.currentValue = null;
                 }
             },
         },
-        textEnabled() {
-            return this.currentStatus == true;
-        },
-    },
-    methods: {
-        /** Submits a changed value. */
-        setValue(value) {
-            this.$emit("input", value, this.id);
-            this.$emit("change", this.refreshOnChange);
```

Comment:
Removed this line since the `change` event is not handled anywhere and `refreshOnChange` does not exist in this module. Additionally, `this.id` is not used either.